planning/planning_scene_monitor/src/planning_scene_monitor.cpp

```diff
@@ -634,6 +636,11 @@ void planning_scene_monitor::PlanningSceneMonitor::excludeRobotLinksFromOctree()
       if (h)
         link_shape_handles_[links[i]].push_back(std::make_pair(h, j));
     }
+    if(!warned && ((ros::Time::now() - start) > ros::Duration(30.0)))
+    {
+      ROS_WARN_STREAM_NAMED("PlanningSceneMonitor::excludeRobotLinksFromOctree()", "It is likely there are too many vertices in collision geometry");
+      warned = true;
```

Comment:
It looks like the "exclusion" of shapes is what takes very long, not the initial loading.
If we want to have a time based measure in the code, this is the only place where it can happen.